lib/src/selection.rs

```diff
@@ -1,122 +1,83 @@
+    // extract all key/value pairs in DOM node
+    props.into_iter().for_each(|(k, v)| {
+        selection.insert(k.to_string(), Value::String(v.to_string()));
+    });
+    trace!(
+        "[{}] got key/value props on a DOM node: {:?}",
+        url,
+        selection
+    );
 
-            other: HashMap::new(),
-        }
+    // text and html
+    let text = el.text().collect::<String>();
+    if text.len() > 0 {
+        selection.insert(String::from("text"), json!(text.trim().to_string()));
+    }
+    if !el.inner_html().is_empty() {
+        selection.insert(
+            String::from("html"),
+            json!(el.inner_html().trim().to_string()),
+        );
     }
-}
-
-pub fn get_selection(el: ElementRef, url: &Url) -> Selection {
-    let mut selection = Selection::new();
 
-    selection.id = elements::id(&el);
-    selection.class = elements::class(&el);
-    selection.style = elements::style(&el);
-    selection.text = elements::text(&el);
-    selection.html = elements::html(&el);
-    selection.href = [elements::href(&el), elements::href_only_child(&el)]
-        .into_iter()
-        .flatten()
-        .next();
-    selection.full_href = match &selection.href {
-        Some(href) => {
-            if href.starts_with("http") {
-                Some(href.to_string())
+    // synthesize other props conditionally
+
+    let image_formats = ["gif", "jpg", "jpeg", "avif", "webp", "ico", "tiff"];
+    // let doc_formats = ["doc", "docx", "pdf"];
+    // let data_formats = ["csv", "json", "txt", "xls"];
+    // let code_formats = ["js", "wasm"];
+    // let style_formats = ["css", "scss", "sass"];
+
+    if let Some(Value::String(href)) = selection.get("href") {
+        if href.starts_with("http") {
+            let href = href.to_string();
+            selection.insert(String::from("hrefType"), json!("absolute"));
+            selection.insert(String::from("full_href"), json!(href));
+        } else {
+            let href = href.to_string();
+            let url = url.join(&href).expect("full url should be parsable");
+            if href.len() == 0 {
+                selection.insert(String::from("hrefType"), json!("empty"));
             } else {
-                let url = url.join(href).expect("full url should be parsable");
-
-                Some(format!("{}", url))
+                selection.insert(String::from("hrefType"), json!("relative"));
+                selection.insert(String::from("full_href"), json!(format!("{}", url)));
             }
         }
-        _ => None,
-    };
+        trace!("[{}] added a full_href prop", url);
+    }
 
-    selection.name = elements::name(&el);
-    selection.content = elements::content(&el);
-    selection.rel = elements::rel(&el);
-    selection.src = elements::src(&el);
-    selection.type_ = elements::type_(&el);
-    selection.disabled = elements::disabled(&el);
+    if let Some(Value::String(src)) = selection.get("src") {
+        let ext = src.split(".").last();
```

Comment:
```suggestion
        let ext = src.split('.').last();
```
char instead of string is enough